File: frontend/src/models/Batch.ts
export class Batch {
  id: number;
  addedDate: Date;
  productId: number;
  quantity: number;
  unitPrice: number;
  expirationDate?: Date;

  constructor(
    id: number,
    addedDate: Date,
    productId: number,
    quantity: number,
    unitPrice: number,
    expirationDate?: Date
  ) {
    this.id = id;
    this.productId = productId;
    this.quantity = quantity;
    this.unitPrice = unitPrice;
    this.addedDate = addedDate;
    this.expirationDate = expirationDate;
  }
}

export class BatchAddDto {
  productId: number;
  quantity: number;
  unitPrice: number;
  expirationDate: string | null = null

  constructor(
    productId: number,
    quantity: number,
    unitPrice: number,
    expirationDate: string | null = null
  ) {
    this.productId = productId;
    this.quantity = quantity;
    this.unitPrice = unitPrice;
    this.expirationDate = expirationDate;
  }
}

File: backend/src/main/java/be/uantwerpen/fti/se/imagineframe_backend/model/Batch.java
package be.uantwerpen.fti.se.imagineframe_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "imf_batches")
public class Batch {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    private int quantity;

    private double unitPrice;

    private LocalDateTime addedDate = LocalDateTime.now();

    private LocalDateTime expirationDate;

    public Batch(Product product, int quantity, double unitPrice) {
        this.product = product;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
    }

    public Batch(Product product, int quantity, double unitPrice, LocalDateTime expirationDate) {
        this.product = product;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
        this.expirationDate = expirationDate;
    }


    public long getProductId() {
        return product.getProductId();
    }
}
